scipy/interpolate/interpolate.py

```diff
@@ -497,10 +499,6 @@ def fill_value(self):
     def fill_value(self, fill_value):
         # extrapolation only works for nearest neighbor and linear methods
         if _do_extrapolate(fill_value):
-            if self._kind not in ('nearest', 'linear'):
-                raise ValueError("Extrapolation does not work with "
-                                 "kind=%s" % self._kind)
-
```

Comment:
Have you checked what happens with a bad input now? Like fill_value="foo"? I think it gets checked by the setter (I might have written that actually!) but it would be good to check